### PR TITLE
fix(ui): history error reported in the dev environment

### LINKED ISSUE #20132: An error occurs when staying on the pod log page
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

 
 when I use the Firefox browser and get stuck on the pod log page, an error occurs in development mode.

**To Reproduce**

**Expected behavior**

display the log normally

**Screenshots**

<!--->

https://github.com/user-attachments/assets/209e723d-6226-4a0c-a2fb-7906d7e90c3e

**Version**

```shell
  "Version": "v2.12.3+6b9cd82",
  "BuildDate": "2024-08-27T11:57:48Z",
  "GitCommit": "6b9cd828c6e9807398869ad5ac44efd2c28422d6",
  "GitTreeState": "clean",
  "GoVersion": "go1.22.4",
  "Compiler": "gc",
  "Platform": "linux/amd64",
  "KustomizeVersion": "v5.4.2 2024-05-22T15:19:38Z",
  "HelmVersion": "v3.15.2+g1a500d5",
  "KubectlVersion": "v0.29.6",
  "JsonnetVersion": "v0.20.0",
  "firefox": 129.0.2 (64-bit)

```

**Logs**

```
Stacktrace:

setWithQueryParams@webpack-internal:///./src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx:111:13
```

### PR DESCRIPTION
**Pull request by youhonglian**
close https://github.com/argoproj/argo-cd/issues/20132

https://github.com/user-attachments/assets/6399168f-d2ca-41ed-a047-e52fa04cf3d2

This is a before video.

https://github.com/user-attachments/assets/cba355d8-0831-47ae-a0a5-f2d0947973b9

This is an after video

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by linghaoSu — APPROVED** [review]
LGTM! Thank you for correcting my fault. It seems I have forgotten to delete redundant code here 😓

Maybe you can change the pr title to  start with `fix(ui):`

**Review by nitishfy — CHANGES_REQUESTED** [review]
Please sign the commits for DCO check to get passed.

**Comment by youhonglian** [comment]
> Please sign the commits for DCO check to get passed.

@nitishfy  Thank you for your reminder, I have signed the commit

**Comment by youhonglian** [comment]
> LGTM! Thank you for correcting my fault. It seems I have forgotten to delete redundant code here 😓
> 
> Maybe you can change the pr title to start with `fix(ui):`

@linghaoSu Thank you for your reminder , I have changed the PR title

**Review by nitishfy — COMMENTED** [review]
Overall LGTM. Can you paste the before vs after here?

**Comment by youhonglian** [comment]
> Overall LGTM. Can you paste the before vs after here?

https://github.com/user-attachments/assets/6399168f-d2ca-41ed-a047-e52fa04cf3d2

This is a before video.

https://github.com/user-attachments/assets/cba355d8-0831-47ae-a0a5-f2d0947973b9

This is an after video

File: ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx [hunk 1]
```diff
@@ -91,8 +91,6 @@ export const PodsLogsViewer = (props: PodLogsProps) => {
     const logsContainerRef = useRef(null);
 
     const setWithQueryParams = <T extends (val: any) => void>(key: string, cb: T) => {
-        history.replaceState(null, '', `${location.pathname}?${queryParams}`);
-
```
Comment (crenshaw-dev) [inline ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx:95]: Do you have any sense why this was present in the first place? Can we be confident that it's not needed?

Comment (crenshaw-dev) [reply ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx:95]: Ah nevermind the original author gave it the thumbs-up. @linghaoSu how far back do we need to cherry-pick this?

Comment (linghaoSu) [reply ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx:95]: I have checked the commit this bug introduced, this change does not ship in any RC or stable version. We don't need to cherry-pick, merging this into the master branch is enough. 

image

Comment (todaywasawesome) [reply ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx:95]: Solves it for me.

**Review by todaywasawesome — APPROVED** [review]
Good to go

**Review by nitishfy — APPROVED** [review]

**Review by pasha-codefresh — APPROVED** [review]

**Comment by pasha-codefresh** [comment]
@crenshaw-dev are you fine with merging this ?

**Pull request merged**